core/nixpkgs.bzl

```diff
@@ -668,3 +670,224 @@ def nixpkgs_package(
+            attr = "nixopts",
+        )
+        for opt in repository_ctx.attr.nixopts
+    ])
+
+    nix_path = executable_path(
+        repository_ctx,
+        "nix",
+        extra_msg = "See: https://nixos.org/nix/",
+    )
+    nix_build = [nix_path, "build"] + expr_args
+
+    # Large enough integer that Bazel can still parse. We don't have
+    # access to MAX_INT and 0 is not a valid timeout so this is as good
+    # as we can do. The value shouldn't be too large to avoid errors on
+    # macOS, see https://github.com/tweag/rules_nixpkgs/issues/92.
+    timeout = 8640000
+    repository_ctx.report_progress("Building Nix derivation")
+    exec_result = execute_or_fail(
+        repository_ctx,
+        nix_build,
+        failure_message = "Cannot build Nix Flake '{}'.".format(nix_build_target),
+        quiet = repository_ctx.attr.quiet,
+        timeout = timeout,
+    )
+    output_path = exec_result.stdout.splitlines()[-1]
+
+    # ensure that the output is a directory
+    test_path = repository_ctx.which("test")
+    execute_or_fail(
+        repository_ctx,
+        [test_path, "-d", output_path],
+        failure_message = "Nix Flake '{}' outputs a single file which is not supported by rules_nixpkgs. Please only use directories.".format(
+            nix_build_target,
+        ),
+    )
+
+    # Build a forest of symlinks (like new_local_package() does) to the
+    # Nix store.
+    for target in find_children(repository_ctx, output_path):
+        basename = target.rpartition("/")[-1]
+        repository_ctx.symlink(target, basename)
+
+    # Create a default BUILD file only if it does not exists and is not
+    # provided by `build_file` or `build_file_content`.
+    if create_build_file_if_needed:
+        p = repository_ctx.path("BUILD")
+        if not p.exists:
+            repository_ctx.template("BUILD", Label("@rules_nixpkgs_core//:BUILD.bazel.tpl"))
+
+_nixpkgs_flake_package = repository_rule(
+    implementation = _nixpkgs_flake_package_impl,
+    attrs = {
+        "nix_flake_file": attr.label(mandatory = True, allow_single_file = ["flake.nix"]),
+        "nix_flake_lock_file": attr.label(mandatory = True, allow_single_file = ["flake.lock"]),
+        "nix_flake_file_deps": attr.label_keyed_string_dict(),
+        "package": attr.string(doc = "Defaults to `default`"),
+        "build_file": attr.label(),
+        "build_file_content": attr.string(),
+        "nixopts": attr.string_list(),
+        "quiet": attr.bool(),
+        "fail_not_supported": attr.bool(default = True, doc = """
+            If set to True (default) this rule will fail on platforms which do not support Nix (e.g. Windows). If set to False calling this rule will succeed but no output will be generated.
+                                        """),
+    },
+)
+
+def nixpkgs_flake_package(
+        name,
+        nix_flake_file,
+        nix_flake_lock_file,
+        nix_flake_file_deps = [],
+        package = None,
```

Comment:
Would `attribute_path` be a better name than `package`?  That would be consistent with `nixpkgs_package()`.  I chose `package` here initially as it matches the name of one of the known Flake output fields.